services/content/service.go

```diff
@@ -356,7 +356,7 @@ func (s *service) Write(session api.Content_WriteServer) (err error) {
 		// users use the same writer style for each with a minimum of overhead.
 		if req.Expected != "" {
 			if expected != "" && expected != req.Expected {
-				return status.Errorf(codes.InvalidArgument, "inconsistent digest provided: %v != %v", req.Expected, expected)
+				log.G(ctx).Debugf("commit digest differs from writer digest: %v != %v", req.Expected, expected)
```

Comment:
What's the reason for these not being errors anymore?

Reply:
Because you can commit using any size or digest that you want. There is nothing in the interface specification that says the committed digest and size MUST match the originally specified digest/size from the Writer. The only thing commit guarantees is that if the size or digest is provided, the content is verified against those before committing.

For use cases, this is need to support append use cases. Where a writer is requested from a given digest but more content is added before commit.